include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_for.h

```diff
@@ -200,13 +200,36 @@ struct __parallel_for_large_submitter<__internal::__optional_kernel_name<_Name..
     }
 };
 
+template <typename _Brick>
+struct __has_pfor_brick_members
+{
+  private:
+    template <typename _F>
+    static auto
+    test(int) -> decltype(std::bool_constant<_F::__can_vectorize>{},
```

Comment:
As far as I understand here you just check that `bool __can_vectorize` and `bool __can_process_multiple_iters` exists but doesn't check that they are declared as `constexpr`.

Reply:
It does check that the members can be evaluated at compile time as the substitution will fail if the members cannot be passed to the `bool_constant` template.

If the compiler can figure out the members at compile time, then it will still work if the members are `const static bool`. In my opinion, we should recommend the members be `constexpr` in the error static_assert, so it's guaranteed to be known at compile time. Is there a way to determine if a variable is declared `constexpr`?

Reply:
I have no idea about such a way because both `constexpr` variable and `const` variable that is initialized by a constant expression can be used in other constant expressions. Thus, to me it looks undistinguishable.